### PR TITLE
Export common traits in `rand` in `ark_std`

### LINKED ISSUE #20: Export `rand::Rng` from `ark_std::rand`
This is related to a number of dependabot's ~pledges~ petitions to bump up the version of rand.
https://github.com/arkworks-rs/nonnative/pull/22
https://github.com/arkworks-rs/r1cs-std/pull/31
https://github.com/arkworks-rs/utils/pull/15
https://github.com/arkworks-rs/gm17/pull/8
https://github.com/arkworks-rs/algebra/pull/151
https://github.com/arkworks-rs/groth16/pull/19

It would be easy if all these repos use `ark_std::rand:...` for the needed traits.

In addition, it would be great if we can handle the `rand_xorshift` in `r1cs_std` and `algebra` as well. If we are likely to keep them, it may be useful to import them here as well (since a majority of the repos would use `algebra` anyway).

### PR DESCRIPTION
**Pull request by weikengchen**
This is a draft PR. More changes that handle #20 would appear soon.

Note: It closes #20

### CODE REVIEW

**Comment by weikengchen** [comment]
The current change is to export the interfaces for 0.7 and 0.2, both old versions.

There will be three steps:
(1) merge this.
(2) modify the downstream to use this instead of rand.
(3) bump the version up from here.

**Comment by weikengchen** [comment]
@Pratyush I don't have permission to this repo and need you to take a look.

By the way, this repo does not have many nice GitHub properties, such as the lovely labels or reviewers.

File: std/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,9 @@ edition = "2018"
 
 [dependencies]
 rand = { version = "0.7", default-features = false }
+# rand = { version = "0.8", default-features = false, features = ["std_rng"] }
+rand_xorshift = "0.2"
+# rand_xorshift = "0.3"
```
Comment (Pratyush) [inline std/Cargo.toml:19]: Do we want to remove these comments for the time being?

File: std/src/rand.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
-use rand::{
-    distributions::{Distribution, Standard},
-    Rng,
-};
+use rand::distributions::{Distribution, Standard};
+
+pub use rand::{distributions::Uniform, rngs::StdRng, Rng, RngCore, SeedableRng};
```
Comment (Pratyush) [inline std/src/rand.rs:3]:
```suggestion
pub use rand;

use rand::rngs::StdRng;
```

File: std/Cargo.toml (hunk 1 above)
Comment (weikengchen) [reply std/Cargo.toml:19]: I probably will keep it---as they would be activated once the downstream are updated.

File: std/src/rand.rs [hunk 3]
```diff
@@ -1,7 +1,9 @@
-use rand::{
-    distributions::{Distribution, Standard},
-    Rng,
-};
+use rand::distributions::{Distribution, Standard};
+
+pub use rand;
+
+use rand::rngs::StdRng;
```
Comment (Pratyush) [inline std/src/rand.rs:5]:
```suggestion
use rand::{Rng, distributions::{Distribution, Standard}, rngs::StdRng};

pub use rand;
```

**Comment by weikengchen** [comment]
Passed the CI.

**Pull request merged**